docs/design/simplified_federation_deployment.md

```diff
@@ -0,0 +1,388 @@
+federation users have to write a yaml file describing a cluster
+resource. They have to specify the cluster's endpoint in the right
+format. That field is too restrictive and hard to get it right. They
+also have to correctly specify where the credentials for the cluster is
+stored.
+
+Therefore, the goal here is to simplify both the federation control
+plane deployment experience and the cluster registration and
+deregistration process. Each of these processes will be reduced to a
+single command with additional flags for customization.
+
+
+## User Experience
+
+Kubernetes v1.4 shipped a new command line tool called
+[`kubeadm`](http://kubernetes.io/docs/getting-started-guides/kubeadm/).
+`kubeadm` simplifies Kubernetes control plane deployment. Federation
+control plane deployment builds on top of this tool. Cluster
+registration and deregistration will be facilitated through
+[`kubectl`](http://kubernetes.io/docs/user-guide/kubectl-overview/).
+
+
+#### Federation Control Plane Deployment Experience
+
+1. User provisions their cloud/compute resources: a VM or a physical
+   machine.
+2. Downloads the
+   [`kubeadm`](http://kubernetes.io/docs/getting-started-guides/kubeadm/)
+   executable on to each of those machines, virtual or physical.
+3. Runs:
+
+    ```shell
+    $ kubeadm init federation
+    ```
+
+   to set up a Kubernetes control plane and a federation control plane
+   on that machine.
+
+
+#### Cluster Registration Experience
+
+1. User downloads the
+   [`kubectl`](http://kubernetes.io/docs/getting-started-guides/kubeadm/)
+   executable on their local computer, i.e. the computer they use for
+   their day-to-day activities.
+2. Runs:
+
+    ```shell
+    $ kubectl register mycluster
+    ```
+
+   to register their cluster named `mycluster`, where `mycluster` is the
+   name of the cluster's context in the local kubeconfig.
+
+#### Cluster Deregistration Experience
+
+1. To deregister a cluster, user runs:
+
+    ```shell
+    $ kubectl deregister mycluster
+    ```
+
+### Points for Discussion:
+
+1. Is `unregister` a better verb than `deregister`?
+2. Does the verb pair `join/unjoin` make more sense in this context?
+
+
+## Design
+
+
+#### Federation Control Plane Bootstrap
+
+`kubeadm init` initializes a Kubernetes control plane. In order to do
+that, it generates the manifests for the various control plane
+components and starts them. It also generates the certificates and the
+credentials required for those components to interact with the API
+server.
+
+Federation control plane deployment builds on top of this functionality.
+Running `kubeadm init federation` first performs all the operations
+involved in bootstrapping a Kubernetes control plane, as running
+`kubeadm init` would. We henceforth call this Kubernetes cluster for
+which the control plane was initialized, the "bootstrap cluster". Upon
+initializing the bootstrap cluster, we perform the following operations
+to bootstrap a federation control plane.
```

Comment:
I have a slightly different mental model that what this seems to suggest.  I think of three possible logical operations

1. Deploy a kubernetes control plane
2. Deploy a federation control plane (into one or more existing kubernetes clusters)
3. Deploy a kubernetes control plane, and automatically deploy a federation control plane into the resulting cluster, once it's ready.

It sounds like you're considering 3 but not 2?